### ISSUE TITLE
issue of a-softmax loss

### ISSUE THREAD

**Issue by chichan01** [wy1iu/sphereface#10]
Hi,
   I set the random_seed in solver.prototxt, use the same CNN architecture and the same training set for evaluating the repeatability.
In softmax loss or softmax with centre loss, I can get the same softmax loss and same center loss in every iteration during training, but it is not the case in a-softmax (sphereface).   In a-softmax, the first iteration is same, but it becomes difference when I repeat to train the model from scratch.  
  Would you like to explain it?

**Comment by wy1iu** [comment]
The loss value varies in each iterations due to different initializations. One possible answer to your question is that A-Softmax loss (with large m, e.g. 4) is more difficult to optimize than center loss, so the loss surface may be much more "non-convex" than the center loss. 

However, while the iteration goes, the variance of the final performance will be much lower. Thus it does not really matter whether each iteration remains the same as the previous training.

**Comment by chichan01** [comment]
Well, I do not agree your point as the initialization of the parameters have been fixed by using random seed and therefore the softmax loss value under no matter center or a-softwax of the first iteration should be same.
